CrossRef.js

```diff
@@ -269,28 +269,32 @@ function processCrossRef(xmlOutput) {
 		var year = ZU.xpathText(pubDateNode[0], 'c:year', ns);
 		var month = ZU.xpathText(pubDateNode[0], 'c:month', ns);
 		var day = ZU.xpathText(pubDateNode[0], 'c:day', ns);
-		
+
 		if(year) {
 			if(month) {
 				if(day) {
 					item.date = year+"-"+month+"-"+day;
 				} else {
-					item.date = month+"/"+year
+					item.date = month+"/"+year;
 				}
 			} else {
 				item.date = year;
 			}
 		}
 	}
-	
+
 	var pages = ZU.xpath(refXML, 'c:pages[1]', ns);
 	if(pages.length) {
 		item.pages = ZU.xpathText(pages, 'c:first_page[1]', ns);
 		var lastPage = ZU.xpathText(pages, 'c:last_page[1]', ns);
 		if(lastPage) item.pages += "-"+lastPage;
 	}
-	
+
 	item.DOI = ZU.xpathText(refXML, 'c:doi_data/c:doi', ns);
+	//add DOI to extra for unsupprted items
+	if (item.DOI && item.itemType != "journalArticle" && item.itemType != "conferencePaper") {
```

Comment:
I like the (more general) function `ZU.fieldIsValidForType("DOI", item.itemType)` which you used in another translator. How about using it here as well?